### PR TITLE
Actionhero v27 + Typed Action tests

### PR DESCRIPTION
**Pull request by evantahler**
Typed `specHelper` from https://github.com/actionhero/actionhero/releases/tag/v27.0.0

### CODE REVIEW

File: core/__tests__/actions/profiles.ts [hunk 1]
```diff
@@ -369,31 +377,30 @@ describe("actions/profiles", () => {
...
         expect(groups.length).toBe(0);
       });
     });
 
     describe("search", () => {
-      let mario: Profile;
-      let luigi: Profile;
-      let peach: Profile;
-      let toad: Profile;
+      let mario: AsyncReturnType<ProfileCreate["run"]>["profile"];
```
Comment (pedroslopez) [inline core/__tests__/actions/profiles.ts]: aha! we found some wrong types! 👍🏼

File: core/src/actions/swagger.ts [hunk 2]
```diff
@@ -120,12 +138,17 @@ export class Swagger extends Action {
...
-      host: url.replace("http://", "").replace("https://", ""),
-      basePath: `/api/v${API_VERSION}`,
-      tags: tags.map((tag) => {
-        return { name: tag };
-      }),
-      schemes: url.includes("https") ? ["https"] : ["http"],
+      host: process.env.WEB_URL
+        ? process.env.WEB_URL.replace("https://", "").replace("http://", "")
+        : "localhost",
+      basePath: `/api/${API_VERSION}`,
```
Comment (pedroslopez) [inline core/src/actions/swagger.ts:145]: Does this change what swagger thinks our API endpoints are? (`basePath` changed `/v1/`->`/1/`)

**Review by pedroslopez — COMMENTED** [review]
This was a big one! Types all around!

Side thought: The use of types like `AsyncReturnType<AppView["run"]>["app"];` is interesting, particularly when being used inside plugins. Maybe if we use these a lot we could type out `AppApiData` or something like that for each of our models and use that. Not a huge deal though, current usage isn't too bad.

File: core/__tests__/actions/profiles.ts (hunk 1 above)
Comment (evantahler) [reply core/__tests__/actions/profiles.ts]: Yeah! That's why I ended up with `AsyncReturnType<...` like you called out above.  But... using AppApiData is much cleaner! I'll update those.

File: core/src/actions/swagger.ts (hunk 2 above)
Comment (evantahler) [reply core/src/actions/swagger.ts:145]: Simplified and cleaned up in 686fe9b2a8aeb3c2d8a4fb00d19746e6c78dac55

**Pull request merged**